include/openthread/platform/infra_if.h

```diff
@@ -133,6 +133,39 @@ extern void otPlatInfraIfRecvIcmp6Nd(otInstance *        aInstance,
  */
 extern otError otPlatInfraIfStateChanged(otInstance *aInstance, uint32_t aInfraIfIndex, bool aIsRunning);
 
+/**
+ * This method sends an address look-up to the well-known host name "ipv4only.arpa" to discover
+ * the NAT64 prefix on the infrastructure interface with @p aInfraIfIndex.
+ *
+ * It's expected that OpenThread calls this method periodically to monitor the presence or change of
+ * NAT64 prefix.
+ *
+ * @param[in]  aInfraIfIndex  The index of the infrastructure interface to discover the NAT64 prefix.
+ *
+ * @retval  OT_ERROR_NONE    Successfully request address look-up.
+ * @retval  OT_ERROR_FAILED  Failed to request address look-up.
+ *
+ */
+otError otPlatInfraIfDiscoverNat64Prefixes(uint32_t aInfraIfIndex);
+
+/**
+ * The infra interface driver calls this method to notify OpenThread that the discovery of NAT64 prefix
+ * is done.
+ *
+ * This method is expected to be invoked after calling otPlatInfraIfDiscoverNat64Prefixes.
+ * If no NAT64 prefix is discovered, this method is expected to be invoked with an empty array @p aIp6Addresses.
+ *
+ * @param[in]  aInstance      The OpenThread instance structure.
+ * @param[in]  aInfraIfIndex  The index of the infrastructure interface on which the address is received.
+ * @param[in]  aIp6Addresses  A pointer to an array of IPv6 addresses synthesized from NAT64 prefix.
+ * @param[in]  aNumAddresses  Number of addresses in array @p aIp6Addresses.
+ *
+ */
+extern void otPlatInfraIfDiscoverNat64PrefixesDone(otInstance *        aInstance,
+                                                   uint32_t            aInfraIfIndex,
+                                                   const otIp6Address *aIp6Addresses,
+                                                   uint8_t             aNumAddresses);
```

Comment:
Updated to pass an `otIp6Prefix` so that it can adapt with the Android platform. If no prefix is discovered, it's expected to pass a pointer to an empty `otIp6Prefix`.

Reply:
Thanks, LGTM!